samza-core/src/main/java/org/apache/samza/table/remote/RemoteTable.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.table.remote;
+
+import java.util.List;
+import java.util.Map;
+
+import org.apache.samza.SamzaException;
+import org.apache.samza.container.SamzaContainerContext;
+import org.apache.samza.table.ReadableTable;
+import org.apache.samza.task.TaskContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+
+
+/**
+ * Samza {@link org.apache.samza.table.Table} backed by a remote data store or service.
+ *
+ * Many stream processing applications need to access some kind of remote data stores or RESTful services
+ * for each received stream message. Such operation can be naturally modeled as a stream-table join where
+ * the store/service is the {@link org.apache.samza.table.Table}. Such abstraction helps application developers
+ * focus on the business logic and leave the stream-table join to the framework. Example use cases include:
```

Comment:
1. s/use cases/use-cases
Can expand a bit more on example scenarios:
1.1 A notifications-system that sends out emails may require a query to an external database to process its message.
1.2 Augmenting a stream of "page-views" with information from a database of user-profiles